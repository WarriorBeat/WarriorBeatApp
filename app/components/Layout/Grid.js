/**
 * Grid.js
 * Generic Grid Components
 * components/Layout
 */

import React from "react"
import PropTypes from "prop-types"
import { Col as GCol, Row as GRow } from "react-native-easy-grid"
import { Position, Shape } from "components/styles"
import { StyleSheet, ViewPropTypes } from "react-native"

const styles = ({
  center, vPad, hPad, justify, wrap, width, height, size,
}) => StyleSheet.create({
  default: {
    ...Position.center(center, justify),
    ...Position.paddingV(vPad),
    ...Position.paddingH(hPad),
    width: width ? Shape.width(width).width : undefined,
    height: height ? Shape.height(height).height : undefined,
    flexWrap: wrap ? "wrap" : "nowrap",
  },
  row: {
    flex: size || (height ? 0 : 1),
  },
  col: {
    flex: size || (width ? 0 : 1),
  },
})

export const Row = (props) => {
  const { children, style } = props
  const compStyle = styles(props)
  return (
    <GRow {...props} style={[compStyle.default, compStyle.row, style]}>
      {children}
    </GRow>
  )
}

export const Col = (props) => {
  const { children, style } = props
  const compStyle = styles(props)
  return (
    <GCol {...props} style={[compStyle.default, compStyle.col, style]}>
      {children}
    </GCol>
  )
}

const GridTypes = {
  center: PropTypes.string,
  style: ViewPropTypes.style,
  children: PropTypes.node.isRequired,
  hPad: PropTypes.string,
  vPad: PropTypes.string,
  justify: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
}

const GridDefault = {
  center: null,
  style: {},
  hPad: "0%",
  vPad: "0%",
  justify: "center",
  height: null,
  width: null,
}

Row.propTypes = {
  ...GRow.propTypes,
  ...GridTypes,
}

Row.defaultProps = GridDefault

Col.propTypes = {
  ...GCol.propTypes,
  ...GridTypes,
}

Col.defaultProps = GridDefault
