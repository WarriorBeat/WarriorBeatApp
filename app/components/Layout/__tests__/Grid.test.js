/**
 * __tests__/Grid.test.js
 * Component Test
 * tests
 */
/* eslint-disable */

import React from "react"
import { shallow, render } from "react-native-testing-library"
import { Grid } from "react-native-easy-grid"
import { Row, Col } from "../Grid"

test("should render correctly", () => {
  const tree = shallow(
    <Grid>
      <Row />
      <Col />
    </Grid>,
  )
  expect(tree).toMatchSnapshot()
})

test("should be centered with padding", () => {
  const tree = render(
    <Grid>
      <Row center="center" />
      <Row vPad="2%" hPad="2%" />
      <Col center="center" justify="flex-start" />
      <Col vPad="2%" hPad="2%" />
    </Grid>,
  )
  expect(tree).toMatchSnapshot()
})
