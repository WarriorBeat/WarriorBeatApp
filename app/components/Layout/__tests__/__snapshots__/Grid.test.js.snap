// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be centered with padding 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    center="center"
    hPad="0%"
    height="100%"
    justify="center"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "height": 1334,
        "justifyContent": "center",
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 750,
      }
    }
    vPad="0%"
    width="100%"
  />
  <View
    center={null}
    hPad="2%"
    height="100%"
    justify="center"
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "height": 1334,
        "justifyContent": "center",
        "paddingHorizontal": 15,
        "paddingVertical": 26.5,
        "width": 750,
      }
    }
    vPad="2%"
    width="100%"
  />
  <View
    center="center"
    hPad="0%"
    height="100%"
    justify="flex-start"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "flexWrap": "nowrap",
        "height": 1334,
        "justifyContent": "flex-start",
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 750,
      }
    }
    vPad="0%"
    width="100%"
  />
  <View
    center={null}
    hPad="2%"
    height="100%"
    justify="center"
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "column",
        "flexWrap": "nowrap",
        "height": 1334,
        "justifyContent": "center",
        "paddingHorizontal": 15,
        "paddingVertical": 26.5,
        "width": 750,
      }
    }
    vPad="2%"
    width="100%"
  />
</View>
`;

exports[`should render correctly 1`] = `
Object {
  "output": <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Row
      center={null}
      hPad="0%"
      height="100%"
      justify="center"
      style={Object {}}
      vPad="0%"
      width="100%"
    />
    <Col
      center={null}
      hPad="0%"
      height="100%"
      justify="center"
      style={Object {}}
      vPad="0%"
      width="100%"
    />
  </View>,
}
`;
